### ISSUE TITLE
flameshot connects to microsoft servers on startup

### ISSUE THREAD

**Issue by AndreSouzaAbreu** [flameshot-org/flameshot#2770]
Hi, it seems like everytime I run `flameshot` it establishes a tcp connection to
one of Microsoft's servers, as I can see from the output of `netstat -apn`.
(See screnshot below

Is this expected? Why so?

![flameshot](https://user-images.githubusercontent.com/47862859/177903131-10e14837-8ff8-4e06-828a-407e14da7119.jpg)

I may be wrong, but this seems to be weird telemetry...
correct me if I'm wrong, but I'd like to know why this behavior.

Version: v12.1.0
Platform: Linux

**Comment by mmahmoudian** [comment]
> I may be wrong

You definitely are

> this seems to be weird telemetry...

If it was, it would have been weird to send your information to lesser of the evils when Meta and Google are there with easy to use APIs

> it establishes a tcp connection to
one of Microsoft's servers

You did a good investigative job, but you ignored the most obvious step: checking the service behind that IP. I've made it easy and have typed it below. Go and check it out and read the text in the footer:

https://20.201.28.148/

> correct me if I'm wrong, but I'd like to know why this behavior.

Let's before throwing guess and accusing us and pointing fingers in the process, we put some facts together:

**Fact1**: Flameshot at startup checks for latest version from our GitHub. I'm on the phone and can't search the source code for you. I leave this investigative part for yourself. Let me know if you need help in finding it.

**Fact2**: GitHub was bought by Microsoft for $7.5 billion in 2018 [[source](https://news.microsoft.com/2018/06/04/microsoft-to-acquire-github-for-7-5-billion/)]

Now 2+2=? 😉

Now more facts:
- We do not have telemetry
- we respect the privacy of our users
- our code if fully opensource and public with reproducible builds, therefore you can check **everything**
- we and other contributors are working on this free and libre opensource software as volunteers and we don't earn money. We do it for free
- we welcome [all contributions](https://github.com/flameshot-org/flameshot/blob/master/docs/CONTRIBUTING.md)
- we do not accept donations, but we accept and encourage our users to but bounty on the bugs and features they want to resolve:
https://rysolv.com/repos/detail/2ce10839-c350-47b1-89a4-2ee9cf0e0678

**Comment by mmahmoudian** [comment]
@AndreSouzaAbreu I forgot to mention that you can turn of "Automatic check for updates":

![image](https://user-images.githubusercontent.com/390889/177939500-7f755485-195c-4631-8e74-da06fbd5ac3e.png)

**Comment by mmahmoudian** [comment]
The main part that you should start from (important line for you are lines 90-92):
https://github.com/flameshot-org/flameshot/blob/b73608060130b94379f7707825abb7d9b64f8a01/src/core/flameshotdaemon.cpp#L55-L93

Then this is called which depending on your installation method (important line for you are lines 186, 193-197):
https://github.com/flameshot-org/flameshot/blob/b73608060130b94379f7707825abb7d9b64f8a01/src/core/flameshotdaemon.cpp#L181-L199

The line 186 in previous part uses `FLAMESHOT_APP_VERSION_URL` variable which is defined here:
https://github.com/flameshot-org/flameshot/blob/b73608060130b94379f7707825abb7d9b64f8a01/src/CMakeLists.txt#L266

Which itself gets its value from `GIT_API_URL` variable which is defined here:
https://github.com/flameshot-org/flameshot/blob/b73608060130b94379f7707825abb7d9b64f8a01/CMakeLists.txt#L7


I hope this has satisfied your curiosity. I always appreciate people double checking my work and pointing out my mistakes, but what I have low tolerance for is unjustified accusations and finger pointing. Thank you for being curious and thank you for raising your concern and question, but please consider adjusting the tone of your text when all you have is a guess/hunch.

Cheers 🍻

**Comment by AndreSouzaAbreu** [comment]
Thank you for answer and the time you took to explain things.
I had forgotten Microsoft has acquired Github and didn't thought of this possibility.

I understand it must be very upset to develop an privacy-aware application and later
on some random folks show up and start questioning telemtry issues. I didn't want
to offend.

I didn't mean to sound hostil or to "point fingers at you" or accuse anybody.
In my mind, the text didn't have any hostile tone. I tried to pass such impression
by saying I did consider I was wrong (and I was). Sorry for not adjusting the text
properly, I was not able to express it well.
